### PR TITLE
[9.x] Chained forwarding

### PR DESCRIPTION
**Pull request by michael-rubel**
## About
This PR introduces a new version of [LEC](https://github.com/michael-rubel/laravel-enhanced-container) package with code niceties and new features.

Instead of relying on the naming conventions, we can now explicitly define the classes we want to forward, and much to say, we can now chain it. 🔥

## How it works?

We can enable the forwarding and define which class to assign in case of missing methods or properties in the base class.

Define the forwarding:
```php
Forwarding::enable()
    ->from(UserService::class)
    ->to(UserRepository::class)
    ->from(UserRepository::class)
    ->to(UserModel::class);
```

Make a proxy:
```php
$proxy = call(UserService::class);
```

Call an existing method:
```php
$proxy->existingMethod();

// Returns result from the `UserService::class`
```

Call a non-existing method:
```php
$proxy->nonExistingMethod();

// Returns result from the `UserRepository::class`, because of forwarding defined above.
```

Call a non-existing method, but that doesn't exist also in `UserRepository::class`:
```php
$proxy->nonExistingInRepositoryMethod();

// Returns result from the `UserModel::class`, because of chained forwarding.
```

### So, why?
- In the first call, we received the result from `UserService`, because the method exists and it is expected.
- In the next call, the proxy managed to replace the internal call instance with `UserRepository`, because of the forwarding defined before.
- In the final call, the proxy swapped the internal instance to `UserModel`, because the method also missing in `UserRepository` (previous instance).

### Use cases?
Still, I'm too lazy to think where to put a query. I may do that in the model, but then move it to the service if additional logic arrives. Then all calls stay untouched. I can manage forwardings at any time and without relying on naming conventions. Separation of concerns cares. 🙂 

### Caveats
Always take attention to which call instance you are working currently when calling methods, accessing, or assigning properties. If some of them don't exist, the proxy will swap an instance and you'll lose the internal state. I'm thinking about storing a previous instance or warning a developer in such a case, we'll see if it fits reality.

### CODE REVIEW

**Comment by michael-rubel** [comment]
Added state management for the `CallProxy` instance. When using forwarding, if the developer interacts multiple times with the same method/property, but on a different internal instance, the `CallProxy` will throw an exception.

We as well able to refer the previous instance in the `CallProxy`:
```php
$proxy->getInternal(Call::PREVIOUS);
```

**Comment by michael-rubel** [comment]
Property assignment forwarding is removed, cause it doesn't make much sense to forward that. If you need to perform a property assignment, just create a new instance of `CallProxy`. If you'll try to interact with the same property on a different internal instance, the exception will be thrown to avoid working on an unexpected instance which I believe improves the developer experience.

```php
Forwarding::enable()
    ->from(UserService::class)
    ->to(UserRepository::class);

$proxy = call(UserService::class);

$proxy->property = true; // `UserService::class`
$proxy->property; // Returns `true`

$proxy->methodMissingInServiceButExistInRepository(); // `UserRepository::class`.
$proxy->property; // `InstanceInteractionException` is thrown.
```

File: src/Overrides/Container.php [hunk 1]
```diff
@@ -708,7 +708,7 @@ public function get(string $id)
                 throw $e;
             }
 
-            throw new EntryNotFoundException($id, $e->getCode(), $e);
+            throw new EntryNotFoundException($id, is_int($e->getCode()) ? $e->getCode() : 0, $e);
```
Comment (valeriisotnyk) [inline src/Overrides/Container.php:711]: `getCode` - always return `int` ?

Comment (michael-rubel) [reply src/Overrides/Container.php:711]: @valeriisotnyk That was a patch applied from the framework: https://github.com/laravel/framework/pull/42393

Comment (valeriisotnyk) [reply src/Overrides/Container.php:711]: @michael-rubel understand. ok :)

File: src/Helpers/helpers.php [hunk 2]
```diff
@@ -16,9 +16,9 @@
     function call(string|object $class, array $parameters = [], ?string $context = null): CallProxy
     {
         return app(CallProxy::class, [
-            'class'        => $class,
+            'class' => $class,
             'dependencies' => $parameters,
```
Comment (olsza) [inline src/Helpers/helpers.php:20]: Fixing the style
```suggestion
            'class'        => $class,
            'dependencies' => $parameters,
```

File: src/Helpers/helpers.php [hunk 3]
```diff
@@ -16,9 +16,9 @@
...
-            'context'      => $context,
+            'context' => $context,
```
Comment (olsza) [inline src/Helpers/helpers.php:21]: Fixing the style
```suggestion
            'context'      => $context,
```

File: src/Helpers/helpers.php (hunk 2 above)
Comment (olsza) [inline src/Helpers/helpers.php:20]: Fixing the style
```suggestion
            'class'        => $class,
            'dependencies' => $parameters,
```

**Pull request merged**